up42/workflow.py

```diff
@@ -378,9 +382,26 @@ def construct_parameters(
                 input_parameters[data_block_name]["ids"] = scene_ids
                 input_parameters[data_block_name]["limit"] = len(scene_ids)
                 input_parameters[data_block_name].pop("time")
-            elif start_date is not None and end_date is not None:
-                time = f"{start_date}T00:00:00Z/{end_date}T23:59:59Z"
-                input_parameters[data_block_name]["time"] = time
+            elif start_date is not None or end_date is not None:
+                try:
+                    start_date_iso: datetime = datetime.fromisoformat(start_date)  # type: ignore
+                    end_date_iso: datetime = datetime.fromisoformat(end_date)  # type: ignore
+                    if len(end_date) == 10:  # type: ignore
+                        # End date provided in the form "2014-01-01" without clock time, is set
+                        # to the previous day as upper boundary of archive.
+                        end_date_iso = datetime.combine(
+                            end_date_iso.date(), datetime_time(23, 59, 59)
+                        )
```

Comment:
We only change the end_date time because it's not set correctly by the datetime.fromisoformat method?

Reply:
The default "todatetime" parsing from a "YYYY-MM-DD" string sets it to the beginning of the day (00:00:00), but the image query requires the end of day. Thanks for the note here, even I got confused now. Overhauled this whole section and clarified it, still not the nicest code but better than before.

Reply:
Looks great now!! Thanks